rfcs/001-mid-level-file-api.md

```diff
@@ -0,0 +1,187 @@
+# Mid-Level File Api
+
+## Summary
+
+This is a mid-level API wrapper around the various File related JavaScript apis. The possibility for a higher level API is left open.
+
+The mid-level file API aims to implement File I/O on top of the raw JS apis found here: https://w3c.github.io/FileAPI/ which includes listing files, creating files, and reading files.
+
+## The API
+
+First we have `FileList` which is simply a list of `File`s. This is an opaque struct that offers a way to iterate over `Files` and get individual files by index:
+
+```rust
+#[derive(Debug, Clone)]
```

Comment:
All of these structs should probably also impl `PartialEq` and `Eq`.

Reply:
`web_sys::FileList` doesn't implement either `PartialEq` or `Eq` so we can't derive it. What would a correct impl for `FileList`?

Reply:
They will implement them in the future though. So I guess we can defer this until then.

(The correct impl would just do the same thing JS does, using `===` or `Object.is` to compare by reference).